src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -2783,10 +2783,28 @@ impl Step for Crate {
             _ => panic!("can only test libraries"),
         };
 
+        let crates = self
+            .crates
+            .iter()
+            .cloned()
+            .map(|crate_| {
+                // The core and alloc crates can't directly be tested. We could
+                // silently ignore them, but replacing them with their test crate
+                // is less confusing for users.
```

Comment:
Is it still possible to run doc-tests now? If `./x test library/core` silently aliases `./x test library/coretests`, then it becomes impossible to run the doctests in libcore...

Reply:
Right, that should probably be handled using a separate doctest only `cargo test` invocation. `cargo test -p core -p alloc` will override the `test = false` in their respective Cargo.toml. So `cargo test -p core -p alloc --doc` would be necessary to only test doctests I think.

Reply:
This has been fixed in https://github.com/rust-lang/rust/pull/137679.